js/utils.js

```diff
@@ -0,0 +1,23 @@
+Fliplet.Registry.set('dynamicListUtils', function() {
+  function dataContains(data, value) {
+    if (!data) {
+      return false;
+    }
+
+    if (_.isArray(data)) {
+      return _.some(data, function (el) {
+        return el.toLowerCase().indexOf(value) > -1;
```

Comment:
what if `el` is not a string?

Reply:
here you should have called the same function in a recursive way instead of redefining the logic

Reply:
https://github.com/Fliplet/fliplet-widget-dynamic-lists/pull/137/commits/45ea25a11add4479bb11d17c48901f26c8650e32 and now we don't have to apply the fix 6 times ☺️

Reply:
@tonytlwu but you're defining the same logic twice in the same function. Definitely wouldn't pass code quality standards :)

Reply:
As I suggested, recursive is the preferred way to handle this situation so you avoid duplicate logic and also deal with nested arrays.

Reply:
https://github.com/Fliplet/fliplet-widget-dynamic-lists/pull/137/commits/97dd0a9c0077568f3e5fc7cdd0e992bbc863de3f